kong/db/schema/init.lua

```diff
@@ -270,6 +270,9 @@ Schema.validators = {
   end,
 
   uuid = function(value)
+    if value == "" then
+      return true
+    end
```

Comment:
Looks like this is related to `config.anonymous`. I see you added `is_present` in the plugins to support both `""` and `ngx.null`, I guess to avoid a migration (always a good goal!).

I wonder if we could get rid of these empty strings via `process_auto_fields` instead. Perhaps with a `legacy` annotation?

```
-- in process_auto_fields (for field.auto == false case only)
if field.legacy and field.uuid and output[key] == "" then
  output[key] == ngx.null
end
```

This would always clean up the data once it comes from the DB (it's a bit of a "migration transformation"). The plugins wouldn't need the `is_present` tests and this change to the `uuid` validator wouldn't be necessary (ie, we'd avoid messing up all UUID fields because of `anonymous`). What do you think?